tests/test_mlkit.py

```diff
@@ -185,34 +265,84 @@
 invalid_string_or_none_args = [0, -1, 4.2, 0x10, False, list(), dict()]
 
 
-def check_error(err, err_type, msg=None):
+# For validation type errors
+def check_error(excinfo, err_type, msg=None):
+    err = excinfo.value
     assert isinstance(err, err_type)
     if msg:
         assert str(err) == msg
 
 
-def check_firebase_error(err, code, status, msg):
+# For errors that are returned in an operation
+def check_operation_error(excinfo, code, msg):
+    err = excinfo.value
+    assert isinstance(err, exceptions.FirebaseError)
+    assert err.code == code
+    assert str(err) == msg
+
+
+# For rpc errors
+def check_firebase_error(excinfo, code, status, msg):
+    err = excinfo.value
     assert isinstance(err, exceptions.FirebaseError)
     assert err.code == code
     assert err.http_response is not None
     assert err.http_response.status_code == status
     assert str(err) == msg
 
 
-def instrument_mlkit_service(app=None, status=200, payload=None):
+def instrument_mlkit_service(status=200, payload=None, operations=False, app=None):
+    if not app:
+        app = firebase_admin.get_app()
+    mlkit_service = mlkit._get_mlkit_service(app)
+    recorder = []
+    if operations:
+        mlkit_service._operation_client.session.mount(
+            'https://mlkit.googleapis.com/v1beta1/',
+            testutils.MockAdapter(payload, status, recorder)
+        )
+    else:
+        mlkit_service._client.session.mount(
+            'https://mlkit.googleapis.com/v1beta1/',
+            testutils.MockAdapter(payload, status, recorder)
+        )
+    return recorder
+
+
+def instrument_mlkit_service_multi(statuses, payloads, operations=False, app=None):
     if not app:
         app = firebase_admin.get_app()
     mlkit_service = mlkit._get_mlkit_service(app)
     recorder = []
-    mlkit_service._client.session.mount(
-        'https://mlkit.googleapis.com',
-        testutils.MockAdapter(payload, status, recorder)
-    )
+    if operations:
+        mlkit_service._operation_client.session.mount(
+            'https://mlkit.googleapis.com/v1beta1/',
+            testutils.MockMultiRequestAdapter(payloads, statuses, recorder)
+        )
+    else:
+        mlkit_service._client.session.mount(
+            'https://mlkit.googleapis.com/v1beta1/',
+            testutils.MockMultiRequestAdapter(payloads, statuses, recorder)
+        )
     return recorder
 
 
 class TestModel(object):
     """Tests mlkit.Model class."""
+    @classmethod
+    def setup_class(cls):
+        cred = testutils.MockCredential()
+        firebase_admin.initialize_app(cred, {'projectId': PROJECT_ID})
+        mlkit._MLKitService.POLL_BASE_WAIT_TIME_SECONDS = 0.1  # shorter for test
```

Comment:
Can this be 0?

Reply:
I left it at 0.1 because that means it actually sleeps which is closer to the actual code execution pattern. (I assumed that sleep(0) doesn't actually sleep. Is that wrong?)
If you are worried about total test time, all 104 tests pass in 0.76 seconds on my machine...